runtime/src/bank.rs

```diff
@@ -2305,14 +2328,33 @@ impl Bank {
         }
     }
 
+    // Distribute collected rent fees for this slot to staked validators (excluding stakers)
+    // according to stake.
+    //
+    // The nature of rent fee is the cost of doing business, every validator has to hold (or have
+    // access to) the same list of accounts, so we pay according to stake, which is a rough proxy for
+    // value to the network.
+    //
+    // Currently, rent distribution doesn't consider given validator's uptime at all (this might
+    // change). That's because rent should be rewarded for the storage resource utilization cost.
+    // It's treated differently from transaction fees, which is for the computing resource
+    // utilization cost.
+    //
+    // We can't use collector_id (which is rotated according to stake-weighted leader schedule)
+    // as an approximation to the ideal rent distribution to simplify and avoid this per-slot
+    // computation for the distribution (time: N log N, space: N acct. stores; N = # of
+    // validators):
+    // - rent fee doesn't need to be incentivized for throughput unlike transaction fees
+    // - leader schedule could be manipulated to locate certain validators more often at the
+    //   start of epoch to unfairly earn more rent for the epoch.
+    // Ref: collect_fees
```

Comment:
@rwalker-com I've added comments based on our discussion some days ago. Does this make sense?